### PR TITLE
Rule-1.2-confusing-identifier-naming

### LINKED ISSUE #245: Rule 1.2 part 6 confusing identifier names
This is a new part in this rule. It was moved from recommendation 3.13.

The following table contains some confusing characters. You should be careful when using them as identifiers. And better use other names instead of these identifiers.

Expected | Confusing name | Suggested name
-- | -- | --
0 (zero) | O, D | obj, dgt
1 (one) | I, l | it, ln, line
2 (two) | Z | n1, n2
5 (five) | S | xs, str
6 (six) | e | ex, elm
8 (eight) | B | bt, nxt
n,h | h,n | nr, head, height
rn, m | m,rn | mbr, item

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Added part 6 of rule 1.2
  * Added warn test on it

## Actions checklist
* [X] Implemented Rule, added Warnings
* [X] Added tests on checks
* [ ] Added tests on fixers
* [X] Updated rules-config.json
* [X] Updated available-rules.md

Closes #245

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 1]
```diff
@@ -173,6 +180,24 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
         return namesOfVariables
     }
 
+    /**
+     * Warns that variable have a confusing name
+     */
+    private fun warnConfusingName (variableName: ASTNode) {
+        var warnText = ""
+        when(variableName.text) {
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:188]: I think you can immediately assign `warnText`

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 2]
```diff
@@ -141,6 +143,11 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
                     if (variableName.text.containsOneLetterOrZero() && variableName.text.length > 1) {
                         VARIABLE_NAME_INCORRECT.warn(configRules, emitWarn, isFixMode, variableName.text, variableName.startOffset)
                     }
+                    // check if identifier of a property has a confusing name
+                    if (CONFUSING_IDENTIFIER_NAMES.contains(variableName.text) && !validCatchIdentifier(variableName)
+                            && node.elementType == ElementType.PROPERTY) {
+                        warnConfusingName(variableName)
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:149]: Do we need to check the right side of property to find expected value @acheshkov?

**Review by kentr0w — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:149]: good point, but by for now diktat is not able to work with such tricky things, so let's check only left side by for now

**Review by akuleshov7 — APPROVED** [review]
fix issue found by Denis and lgtm

**Review by kentr0w — APPROVED** [review]

**Review by petertrr — APPROVED** [review]

**Pull request merged**